### PR TITLE
DIT-806: Add Thumbnails to Search & DIT-807: Add Cancellation Reason to Search

### PR DESCRIPTION
**Pull request by ed0906**
Prototype: https://www.prototypes.tax.service.gov.uk/digital-tariffs-operator-ui-prototype/operational-ui-vF/search-results-4

### CODE REVIEW

File: app/uk/gov/hmrc/tariffclassificationfrontend/models/CancelReason.scala [hunk 1]
```diff
@@ -24,13 +24,22 @@ object CancelReason extends Enumeration {
...
+      case INVALIDATED_CODE_CHANGE => s"Invalidated due to nomenclature code changes (${code(INVALIDATED_CODE_CHANGE)})"
+      case INVALIDATED_EU_MEASURE => s"Invalidated due to EU measure (${code(INVALIDATED_EU_MEASURE)})"
+      case INVALIDATED_NATIONAL_MEASURE => s"Invalidated due to national legal measure (${code(INVALIDATED_NATIONAL_MEASURE)})"
+      case INVALIDATED_WRONG_CLASSIFICATION => s"Invalidated due to incorrect classification (${code(INVALIDATED_WRONG_CLASSIFICATION)})"
+      case INVALIDATED_OTHER => s"Invalidated due to other reasons (${code(INVALIDATED_OTHER)})"
       case unknown => throw new IllegalArgumentException(s"Unexpected reason: $unknown")
     }
   }
+
+  def code(reason: CancelReason): Int = reason match {
```
Comment (gokyo) [inline app/uk/gov/hmrc/tariffclassificationfrontend/models/CancelReason.scala]: can be private?

File: app/uk/gov/hmrc/tariffclassificationfrontend/service/FileStoreService.scala [hunk 2]
```diff
@@ -32,15 +32,34 @@ class FileStoreService @Inject()(connector: FileStoreConnector) {
 
   def getAttachments(c: Case)(implicit hc: HeaderCarrier): Future[Seq[StoredAttachment]] = {
     val attachmentsById: Map[String, Attachment] = c.attachments.map(a => a.id -> a).toMap
-    connector.get(c.attachments) map { files =>
-      files map { file =>
+    connector.get(c.attachments) map {
```
Comment (gokyo) [inline app/uk/gov/hmrc/tariffclassificationfrontend/service/FileStoreService.scala]: can this code used below?

looks like the two methods have code in common.... we could extract it and define the common code only once

File: app/uk/gov/hmrc/tariffclassificationfrontend/models/CancelReason.scala (hunk 1 above)
Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/models/CancelReason.scala]: It's used in a view too

File: app/uk/gov/hmrc/tariffclassificationfrontend/service/FileStoreService.scala (hunk 2 above)
Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/service/FileStoreService.scala]: Done

File: app/uk/gov/hmrc/tariffclassificationfrontend/service/FileStoreService.scala [hunk 3]
```diff
@@ -32,15 +32,34 @@ class FileStoreService @Inject()(connector: FileStoreConnector) {
...
+        case (c: Option[Case], att: Option[StoredAttachment]) => c.isDefined && att.isDefined
+      } map {
+        case (c: Option[Case], att: Option[StoredAttachment]) => (c.get, att.get)
+      } groupBy(_._1) map {
+        case (c: Case, seq: Seq[(Case, StoredAttachment)]) => (c, seq.map(_._2))
+      }
+      val missing = cases.filterNot(group.contains)
+      group ++ missing.map(_ -> Seq.empty)
+    }
+  }
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/tariffclassificationfrontend/service/FileStoreService.scala]: That's a busy method.  Looks lovely but I have no idea what the logic's doing and hence whether it's doing the right thing.

Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/service/FileStoreService.scala]: I've attempted to pull it out a bit & add a few comments. The logic is quite involved so I'm not sure how better to do it.

**Pull request merged**